web/packages/teleterm/src/ui/TopBar/NavigationMenu/NavigationMenu.tsx

```diff
@@ -18,75 +18,102 @@ import React, { useState, useRef } from 'react';
 import styled from 'styled-components';
 
 import Popover from 'design/Popover';
-import { MoreVert, OpenBox, Add } from 'design/Icon';
-import { Box, Text, Flex } from 'design';
+import { MoreVert, OpenBox, Add, Config } from 'design/Icon';
 
 import { useAppContext } from 'teleterm/ui/appContextProvider';
-import { DocumentsService } from 'teleterm/ui/services/workspacesService';
 import { TopBarButton } from 'teleterm/ui/TopBar/TopBarButton';
-import { RootClusterUri } from 'teleterm/ui/uri';
-
-import { useIdentity } from '../Identity/useIdentity';
+import { IAppContext } from 'teleterm/ui/types';
+import { Cluster } from 'teleterm/services/tshd/types';
 
 import { NavigationItem } from './NavigationItem';
 
-function getNavigationItems(
-  documentsService: DocumentsService,
-  clusterUri: RootClusterUri
-): {
-  title: string;
-  Icon: JSX.Element;
-  onNavigate: () => void;
-}[] {
+function useNavigationItems(): (
+  | {
+      type?: 'normal';
+      title: string;
+      Icon: JSX.Element;
+      onNavigate: () => void;
+    }
+  | { type: 'separator' }
+)[] {
+  const ctx = useAppContext();
+  ctx.workspacesService.useState();
+  ctx.clustersService.useState();
+
+  const documentsService =
+    ctx.workspacesService.getActiveWorkspaceDocumentService();
+  const activeRootCluster = getActiveRootCluster(ctx);
+  const accessRequestsAllowed =
+    !!activeRootCluster?.features?.advancedAccessWorkflows;
+
   return [
     {
+      title: 'Open Config File',
+      Icon: <Config fontSize={2} />,
```

Comment:
Can we pass a component here and then render it with `fontSize={2}` instead of having to remember to pass it for each icon?